tests/generate/test_generate_client.py

```diff
@@ -0,0 +1,48 @@
+import pathlib
+import shutil
+
+from _pytest.tmpdir import TempPathFactory
+
+from tests.utils.approvals import verify
+from tests.utils.click_invoker import invoke
+
+
+def test_generate_help() -> None:
+    result = invoke("generate -h")
+
+    assert result.exit_code == 0
+    verify(result.output)
+
+
+def test_generate_client_python(tmp_path_factory: TempPathFactory) -> None:
+    cwd = tmp_path_factory.mktemp("cwd")
+    json_file = pathlib.Path(__file__).parent / "application.json"
+    shutil.copy(json_file, cwd / "application.json")
```

Comment:
This looks good, I think one change we can make to improve the test maintainability is to create a fixture that creates the temp directory, copies the application.json into it and then returns the path to the temp application.json

Then all the tests can depend on that fixture